### PR TITLE
Scan: implement send delegate msg

### LINKED ISSUE #1424: Implement UI & Wire up about send `Delegate` msg
- implement UI as a design on #1423 
- implement creating delegate msg like request
- show error properly if balance is the insufficient or invalid input field
- testing with mnemonic and ledger

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #1424

### CODE REVIEW

File: scan/src/components/modal/submitTx/DelegateMsg.re [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      setInputData=setValidator
+      parse=Address.fromBech32Opt
+      msg="Delegate to"
+      errMsg="Invalid Address"
+    />
+    <VSpacing size=Spacing.md />
+    <EnhanceTxInput
+      width=115
+      inputData=amount
+      setInputData=setAmount
```
Comment (tansawit) [inline scan/src/components/modal/submitTx/DelegateMsg.re:37]: Change amount unit from uband to band

File: scan/src/components/modal/submitTx/SubmitTxModal.re [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        </div>
+      </div>
+      <VSpacing size=Spacing.md />
+      {switch (msgType) {
+       | Send => <div />
+       | Delegate => <DelegateMsg setMsgsOpt />
+       | Undelegate => <div />
+       }}
+      <VSpacing size=Spacing.md />
+      // TODO: add gas, fee, memo when create tx later
```
Comment (tansawit) [inline scan/src/components/modal/submitTx/SubmitTxModal.re]: Can we remove this comment now that we have those fields?

File: scan/src/components/modal/submitTx/DelegateMsg.re [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      parse=int_of_string_opt
+      msg="Amount (BAND)"
+      errMsg="Invalid amount"
+    />
```
Comment (tansawit) [inline scan/src/components/modal/submitTx/DelegateMsg.re]: Change unit from uband to band

**Review by tansawit — CHANGES_REQUESTED** [review]

File: scan/src/components/modal/submitTx/DelegateMsg.re (hunk 3 above)
Comment (evilpeach) [reply scan/src/components/modal/submitTx/DelegateMsg.re]: I just change it to `uband` for now.

File: scan/src/components/modal/submitTx/DelegateMsg.re [hunk 4]
```diff
@@ -0,0 +1,43 @@
+[@react.component]
+let make = (~setMsgsOpt) => {
+  let (validator, setValidator) = React.useState(_ => EnhanceTxInput.{text: "", value: None});
```
Comment (sorawit) [inline scan/src/components/modal/submitTx/DelegateMsg.re]: Why dont we create `EnhanceTxInput.empty = {text: "", value: None}` so we can reuse?

File: scan/src/components/modal/submitTx/SubmitTxModal.re [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        parse=int_of_string_opt
+        msg="Fee"
+        errMsg="Invalid amount"
+      />
+      <VSpacing size=Spacing.md />
+      <EnhanceTxInput
+        width=300
+        inputData=memo
+        setInputData=setMemo
+        parse={newVal => {newVal->Js.String.length <= 20 ? Some(newVal) : None}}
```
Comment (sorawit) [inline scan/src/components/modal/submitTx/SubmitTxModal.re]: Let's set maximum at 32.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by tansawit — APPROVED** [review]

**Pull request merged**